MachObfuscator/Controller/Obfuscator.swift

```diff
@@ -48,7 +49,9 @@ class Obfuscator {
         let symbols: ObfuscationSymbols = time(withTag: "Build obfuscation symbols") {
             autoreleasepool {
                 let skippedSymbols = ObjectSymbols.blacklist(skippedSymbolsSources: options.skippedSymbolsSources,
-                                                             sourceSymbolsLoader: sourceSymbolsLoader)
+                                                             skippedSymbolsLists: options.skippedSymbolsLists,
```

Comment:
You are right, but this will also make the `Obfuscator` larger. I think `Obfuscator` needs a little refactor. It is kinda linear, but it grew very long. And it needs unit tests. I will apply your suggestion. I will also consider refactoromg the `Obfuscator` class.

Reply:
I agree, that `Obfuscator` class is quite big but I think that at this point it is still manageable. `Obfuscator` class clearly shows what steps are taken in which order and what data flow between them (so it is indeed a controller). It also decouples classes implementing logic from input arguments which simplifies testing them.

`Obfuscator` class can be tested. These tests however would be rather integration tests than unit tests. I have implemented some tests like this. They need to build `Options` and replicate some code from `main` function but they are not so complicated after all.

I was considering some deeper refactoring of analysis and obfuscation passes, which would be helpful for some Swift obfuscations which are dependent on ObjC obfuscations and the other way around (I had to do some workarounds to bridge them). However, this would be a major architectural change.